webhelper/src/gwh-browser.c

```diff
@@ -950,6 +1005,23 @@ gwh_browser_set_uri (GwhBrowser  *self,
   g_free (real_uri);
 }
 
+gboolean
+gwh_browser_set_uri_from_document (GwhBrowser    *self,
+                                   GeanyDocument *doc)
+{
+  gchar *uri;
+
+  /* document must exist on disk */
+  if (! doc || ! doc->real_path)
+    return FALSE;
+
+  uri = g_strconcat ("file://", doc->file_name, NULL);
```

Comment:
Why test doc->real_path above and use untested doc->file_name here?  Shouldn't this use the path?

Reply:
No, `doc->real_path` is used to verify the document is actually saved to a file, but it's in locale encoding, whereas `doc->file_name` is the expected UTF-8.  Admittedly, it might not behave properly if the locale is non-UTF-8 and the file path contains non-ASCII characters; but at least the worse that would happen would be the file not being found.

Reply:
Ok, I forgot the inconsistency that sometimes paths are called paths and sometimes filenames in C land, ... sigh @elextr goes back to his nice consistent C++ `std::path` :grin: